packages/angular_devkit/build_angular/test/browser/bundle-budgets_spec_large.ts

```diff
@@ -64,6 +64,68 @@ describe('Browser Builder bundle budgets', () => {
     await run.stop();
   });
 
+  it(`shows warnings for large component css when using 'anyComponentStyle' when AOT`, async () => {
+    const overrides = {
+      aot: true,
+      optimization: true,
+      budgets: [{ type: 'anyComponentStyle', maximumWarning: '1b' }],
```

Comment:
What's the plan for showing this warning overall? Will we update bundle budgets in 9 for all projects?

Reply:
I think the best option would be to do it for 9 for existing projects since users won't typically run migrations on minor and for new projects we can do it right away.

1) What should be the value `maximumWarning` and `maximumError` for new projects?
2) I am guessing that for the migration we won't want to add `maximumError` as this will break exiting projects.

Reply:
Yeah I think it should be for 9 too. We should add the migration to this PR as well.

As for the default numbers, I'm not really sure. UTF-8 uses [1 to 4 bytes](https://stackoverflow.com/a/10229225/2116927) for each char, most commonly only 1 byte for US ASCII chars. So 1kb would be 1024 chars. Off the top of my head warning on 5kb and erroring on 20kb sounds within the realm of reasonable. @mgechev do you have a recommendation?

Reply:
Ill start working on the migrations, and than we just change the number when we have more feedback from @mgechev.

Reply:
A good source for setting up good defaults will be http://bit.ly/perf-budget-calculator

I'd recommend bumping the error limit up with few extra 100s of KBs.

Reply:
@mgechev the budgets will be per component.

Reply:
As an example, in https://github.com/johnpapa/awesome-angular-workshop/ several components included 70kb of minified styles each. This happened because they included the whole material theme.

Reply:
Yes, missed the fact it's per component. Budget for component is definitely not a complete solution but sounds like a good start.

Here's a very rough estimation based on https://bit.ly/perf-budget-calculator

100KB from the calculator is very roughly 20 components (totally making this up)
20KB / 20 components = 1KB
Bumping this up with 5KB

~6KB default warning
~10KB default error

Reply:
@alan-agius4 do we have a way forward with a global css bundle budget? Not asking to have it included in this PR, just asking if the changes in this PR would get in the way of that. I imagine a `anyGlobalCss` budget or something.

Reply:
Global styles meaning styles in general or that are set globally ie imported via style.css?

For the later we already have the `AnyCalculator` which can be used to set budgets on global css.

However, if we'd want to add a budget for css within the component + global css we can do it and this PR will provide a way forward for that.

My concerns would be that if want to sum up css from all the components and global stylesheet, you might end up with a misleading budget. Let's say the result of total CSS is 500Kb, when having lazy loading this will never be loaded/parsed at once.

Reply:
Dynamic budgets, such as LightWallet would be great in such scenario. In the same time, we might be able to estimate what would be loaded initially, right? In webpack we have this metadata in the compilation stats.

Reply:
But this is indeed is  the enabler for that :)

Reply:
We already have some budgets for initial chunks https://github.com/angular/angular-cli/blob/47c0db9eb394640501af5f0fd36495547466bc37/packages/angular_devkit/build_angular/src/angular-cli-files/utilities/bundle-calculator.ts#L62